omniNotes/src/androidTest/java/it/feio/android/omninotes/utils/SortNotes.java

```diff
@@ -0,0 +1,30 @@
+package it.feio.android.omninotes.utils;
+
+import androidx.test.espresso.ViewInteraction;
+
+import java.util.ArrayList;
+
+import it.feio.android.omninotes.R;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withParent;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static it.feio.android.omninotes.utils.EspressoUtils.*;
+import static it.feio.android.omninotes.utils.NewNoteStringVariables.titleButton;
+import static org.hamcrest.Matchers.allOf;
+
+public class SortNotes {
+    public static ArrayList<String> sortNotes(ArrayList<String> arrayWithTitles, int numberOfTimes) {
+        int index = 0;
+        ViewInteraction sortButton = onView(withId(R.id.menu_sort));
+
+        clickOnButton(sortButton);
+        clickOnButton(onView(withText(titleButton)));
```

Comment:
when switching between sorting options, i noticed that not all the times the selection takes place. sometimes, you need to redo the action. therefore, it is necessary some kind of check here that confirms the action

Reply:
I changed the method so it will click the button until the selection is made